### PR TITLE
feature unix time. excel time #closes 1420

### PR DESCRIPTION
**Pull request by t-m-z**
feature wish from Nasenbaer69

### CODE REVIEW

**Comment by t-m-z** [comment]
yes I thought about it. 

Regarding https://en.wikipedia.org/wiki/Unix_time Unix time by design does not require a specific size for the storage. Hence there will be no problem in theory.
Unfortunately in the real world sometimes 32bit integers are used to store the time. Hence this year 2038 Problem will arise - https://en.wikipedia.org/wiki/Year_2038_problem
Therefore a modulo and recount could be an option.

**Comment by S-Man42** [comment]
I believe, JS could make problems nonetheless. We already had this in the past: While dart does not have any issues with numbers in that size, JS has a limitation (not quite sure which). But we should have an eye on it. I believe, it's such a crazy thought that somebody may build a computer-based mystery where he might use this range border for something. Not in 12 years. So we should think about how to face it.

I will merge it, but maybe you'll get an idea to deal with it.

**Comment by S-Man42** [comment]
Please add tests

**Comment by t-m-z** [comment]
tests added

**Comment by S-Man42** [comment]
grafik

Irgendwo packt der Sekunden-Nachkommastellen aus. Bitte mal checken, wo das herkommt. Fraglich ist überhaupt, ob Nachkommastellen überhaupt Sinn ergeben.

**Comment by S-Man42** [comment]
Encryption: Maybe it could be a good idee to add a Divider with (year - month -day) desciption. US amis may switch day and month or something.

Decryption: Please return the localized date.

**Comment by S-Man42** [comment]
Negative Unix Time in "Calender"? Does this make any sense? If so, we should be consistent and add it to the UNIX Timestamp Tool as well (in that case please add tests just to demonstrate the feature)

grafik

File: assets/searchstrings/de.json [hunk 1]
```diff
@@ -127,7 +127,7 @@
   "dates_day_of_the_year": "tagesnummern tagnummern tagzahlen tageszahlen tagesanzahlen wochentage wochennummern wochenzahlen wochenanzahlen jahrestage tagimjahr tagdesjahres",
   "dates_timecalculator": "uhrzeit uhrzeitrechner stunden minuten sekunden zeitenrechner",
   "dates_weekday": "wochentage",
-  "dates_calendar": "julianisches datum gregorianisches datum kalender",
+  "dates_calendar": "julianisches datum gregorianisches datum kalender zeitstempel maya koptisch potrzebie tzolkin haab yazdegard",
```
Comment (S-Man42) [inline assets/searchstrings/de.json:130]: As usual, translations only in Crowdin! Otherwise we might get conflicts.

**Comment by t-m-z** [comment]
> grafik
> 
> Irgendwo packt der Sekunden-Nachkommastellen aus. Bitte mal checken, wo das herkommt. Fraglich ist überhaupt, ob Nachkommastellen überhaupt Sinn ergeben.

will be removed with localized date format

**Comment by t-m-z** [comment]
> Encryption: Maybe it could be a good idee to add a Divider with (year - month -day) desciption. US amis may switch day and month or something.
> 
> Decryption: Please return the localized date.

Encyption: this means to enhance the DateTimePicker-Widget according to locale settings

Encryption: done

**Comment by S-Man42** [comment]
"Encyption: this means to enhance the DateTimePicker-Widget according to locale settings"

Dont understand. Whats the problem by simply add a divider widget before the datepicker. There's no reason to touch the date picker...

**Comment by t-m-z** [comment]
> "Encyption: this means to enhance the DateTimePicker-Widget according to locale settings"
> 
> Dont understand. Whats the problem by simply add a divider widget before the datepicker. There's no reason to touch the date picker...

Your point ouches every usage of the DateTimePicker - should I really add a Divider before every usage? W used this widget about 14 times

**Comment by S-Man42** [comment]
hmm... ok we shoukd think about more generel

**Comment by t-m-z** [comment]
done - however I see a strange behaviour with eth Excel-Timestamp. There is a kind of overflow Error I can not really identify.

**Comment by S-Man42** [comment]
> There is a kind of overflow Error I can not really identify.

Where is it? What needs to be done? Maybe leave Excel out for the moment and only insert Unix?

**Comment by S-Man42** [comment]
Please make "Decrypt" mode the initial one.

**Pull request merged**